### PR TITLE
[chip-cert] Updated Tool To Generate Invalid Operational Certificates.

### PR DESCRIPTION
**Pull request by emargolis**
#### Problem
Need APIs to generate invalid conditions when sending AddTrusted, AddNOC, UpdateNOC commands.

Ticket: #19068 

#### Change overview
- Updated current code to inject errors into operational certs in DER/PEM format.
- New code to generate CHIP TLV encoded certificates with injected errors.
- It can generate invalid NOC, ICAC, and RCAC.

#### Testing
Verified manually that the tool generates normal/valid operational certificates as before.
Verified manually that invalid certificates can be properly generated.

### CODE REVIEW

File: src/tools/chip-cert/CertUtils.cpp [hunk 1]
```diff
@@ -310,6 +330,108 @@ bool AddExtension(X509 * cert, int extNID, const char * extStr)
     return res;
 }
 
+bool SetBasicConstraintsExtension(X509 * cert, bool isCA, int pathLen, CertStructConfig & certConfig)
+{
+    if (certConfig.IsExtensionBasicPresent())
```
Comment (andy31415) [inline src/tools/chip-cert/CertUtils.cpp:335]: nit: some indentation would decrease (and readability increase) if you invert conditions:

```cpp
if (!certConfig.IsExtensionBasicPresent()) {
  return true;
}

// ....

return AddExtension(cert, NID_basic_constraints, basicConstraintsExt.c_str());
```

File: src/tools/chip-cert/CertUtils.cpp [hunk 2]
```diff
@@ -310,6 +330,108 @@ bool AddExtension(X509 * cert, int extNID, const char * extStr)
...
+    {
+        std::string basicConstraintsExt;
+
+        if (certConfig.IsExtensionBasicCriticalPresent())
```
Comment (andy31415) [inline src/tools/chip-cert/CertUtils.cpp:339]: use && for these to make condition depth smaller.

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/tools/chip-cert/CertUtils.cpp (hunk 2 above)
Comment (emargolis) [reply src/tools/chip-cert/CertUtils.cpp]: done

File: src/tools/chip-cert/CertUtils.cpp (hunk 1 above)
Comment (emargolis) [reply src/tools/chip-cert/CertUtils.cpp]: done

**Review by msandstedt — APPROVED** [review]

**Pull request merged**